test/Escrow.t.sol

```diff
@@ -381,6 +369,62 @@ contract EscrowTest is AtlasBaseTest {
         executeSolverOperationCase(userOp, solverOps, true, false, result, true);
     }
 
+    function test_executeSolverOperation_solverOpWrapper_BalanceNotReconciled_ifPartialRepayment() public {
+        uint256 bidAmount = dummySolver.partialGasPayBack(); // solver only pays half of shortfall
+        deal(address(dummySolver), bidAmount);
+
+        (UserOperation memory userOp, SolverOperation[] memory solverOps) = executeSolverOperationInit(defaultCallConfig().build());
+        solverOps[0] = validSolverOperation(userOp)
+            .withBidAmount(bidAmount)
+            .signAndBuild(address(atlasVerification), solverOnePK);
+        uint256 expectedResult = (1 << uint256(SolverOutcome.BalanceNotReconciled));
+        executeSolverOperationCase(userOp, solverOps, true, false, expectedResult, true);
+    }
+
+    function test_executeSolverOperation_solverOpWrapper_Success_ifPartialRepaymentAndDappCoversTheRest() public {
+        uint256 bidAmount = dummySolver.partialGasPayBack(); // solver only pays half of shortfall
+        deal(address(dummySolver), bidAmount);
+
+        GasSponsorDAppControl gasSponsorControl = new GasSponsorDAppControl(
+            address(atlas),
+            address(governanceEOA),
+            defaultCallConfig()
+                .withTrackPreOpsReturnData(true)
+                .withTrackUserReturnData(true)
+                .withRequirePreOps(true)
+                .withPostSolver(true)
+                .build());
+    
+        // Give dapp control enough funds to cover the shortfall
+        deal(address(gasSponsorControl), 1 ether);
+
+        vm.prank(governanceEOA);
+        atlasVerification.initializeGovernance(address(gasSponsorControl));
+
+        UserOperation memory userOp = new UserOperationBuilder()
+            .withFrom(userEOA)
+            .withTo(address(atlas))
+            .withValue(0)
+            .withGas(1_000_000)
+            .withMaxFeePerGas(tx.gasprice + 1)
+            .withNonce(address(atlasVerification), userEOA)
+            .withDeadline(block.number + 2)
+            .withDapp(address(gasSponsorControl))
+            .withControl(address(gasSponsorControl))
+            .withCallConfig(gasSponsorControl.CALL_CONFIG())
+            .withSessionKey(address(0))
+            .withData(abi.encodeWithSelector(gasSponsorControl.userOperationCall.selector, false, 0))
+            .signAndBuild(address(atlasVerification), userPK);
+        
```

Comment:
Better to use the default and only override the necessary fields if possible:

```suggestion
        UserOperation memory userOp = validUserOperation()
            .withDapp(address(gasSponsorControl))
            .withControl(address(gasSponsorControl))
            .withCallConfig(gasSponsorControl.CALL_CONFIG())
            .withData(abi.encodeWithSelector(gasSponsorControl.userOperationCall.selector, false, 0))
            .signAndBuild(address(atlasVerification), userPK);
```

This simplifies maintenance in future if the defaults ever need to change, we can update them in fewer places.

Reply:
Had to refactor it a bit more to get it to not try call the default control which hadn't been deployed, to get the call config. But now should be more reusable as it takes any dapp control address. Done here https://github.com/FastLane-Labs/atlas/pull/225/commits/43f7ce74a89eb7b7c838745ad778e4e99b609c02